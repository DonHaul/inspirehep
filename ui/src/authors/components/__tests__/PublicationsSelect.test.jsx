import React from 'react';
import { render } from '@testing-library/react';

import PublicationsSelect from '../PublicationsSelect';

describe('PublicationsSelect', () => {
  it('sets publication selection on checkbox change', () => {
    const onSelectClaimedPapers = jest.fn();
    const onSelectUnclaimedPapers = jest.fn();
    const onSelectPapers = jest.fn();
    const { asFragment, getByRole } = render(
      <PublicationsSelect
        claimed
        isOwnProfile
        onSelectClaimedPapers={onSelectClaimedPapers}
        onSelectUnclaimedPapers={onSelectUnclaimedPapers}
        onSelectPapers={onSelectPapers}
      />
    );
    expect(asFragment()).toMatchSnapshot();
    getByRole('checkbox').click();
    expect(onSelectClaimedPapers).toHaveBeenCalled();
    expect(onSelectPapers).toHaveBeenCalled();
  });

  it('renders checked when selected', () => {
    const onSelectClaimedPapers = jest.fn();
    const onSelectUnclaimedPapers = jest.fn();
    const onSelectPapers = jest.fn();
    const { asFragment } = render(
      <PublicationsSelect
        claimed
        checked
        onSelectClaimedPapers={onSelectClaimedPapers}
        onSelectUnclaimedPapers={onSelectUnclaimedPapers}
        onSelectPapers={onSelectPapers}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders unchecked when not selected', () => {
    const onSelectClaimedPapers = jest.fn();
    const onSelectUnclaimedPapers = jest.fn();
    const onSelectPapers = jest.fn();
    const { asFragment } = render(
      <PublicationsSelect
        claimed
        checked={false}
        onSelectClaimedPapers={onSelectClaimedPapers}
        onSelectUnclaimedPapers={onSelectUnclaimedPapers}
        onSelectPapers={onSelectPapers}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
