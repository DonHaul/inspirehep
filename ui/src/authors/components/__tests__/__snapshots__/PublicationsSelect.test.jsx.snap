// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationsSelect renders checked when selected 1`] = `
<DocumentFragment>
  <label
    class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
  >
    <span
      class="ant-checkbox ant-checkbox-checked"
    >
      <input
        checked=""
        class="ant-checkbox-input"
        type="checkbox"
        value=""
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
  </label>
</DocumentFragment>
`;

exports[`PublicationsSelect renders unchecked when not selected 1`] = `
<DocumentFragment>
  <label
    class="ant-checkbox-wrapper"
  >
    <span
      class="ant-checkbox"
    >
      <input
        class="ant-checkbox-input"
        type="checkbox"
        value=""
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
  </label>
</DocumentFragment>
`;

exports[`PublicationsSelect sets publication selection on checkbox change 1`] = `
<DocumentFragment>
  <label
    class="ant-checkbox-wrapper"
  >
    <span
      class="ant-checkbox"
    >
      <input
        class="ant-checkbox-input"
        type="checkbox"
        value=""
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
  </label>
</DocumentFragment>
`;
